SimG4Core/Application/plugins/OscarProducer.cc

```diff
@@ -212,11 +212,13 @@ void OscarProducer::produce(edm::Event & e, const edm::EventSetup & es)
 
   } catch ( const SimG4Exception& simg4ex ) {
        
-    edm::LogInfo("SimG4CoreApplication") << " SimG4Exception caght !" 
+    edm::LogInfo("SimG4CoreApplication") << "SimG4Exception caght! " 
 					 << simg4ex.what();
        
-    m_runManager->abortEvent();
```

Comment:
This seems to be needed to properly clear Geant's internals in the case of a failure.
Also, for the original problem from the issue, `produce` is never called since the job is ending before entering the event loop.